CanvasPlusPlayground/Views/CourseAnnouncementsView.swift

```diff
@@ -18,17 +18,17 @@ struct CourseAnnouncementsView: View {
     
     var body: some View {
         NavigationStack {
-            List(announcementManager.announcements, id:\.id) { announcment in
+            List(announcementManager.announcements, id:\.id) { announcement in
                 NavigationLink {
-                    CourseAnnouncementDetailView(announcement: announcment)
-                        .onAppear {
-                            announcment.update(keypath: \.isRead, value: true)
+                    CourseAnnouncementDetailView(announcement: announcement)
+                        .task {
```

Comment:
This would update the `isRead` property when the list cell loads, not when the view is clicked on, right?

Reply:
Could you also try using `announcement.isRead = true` and see if this works. It would be annoying if we had to use the `update` function everywhere. I believe we're using it for the course color, etc. and it works fine.

Reply:
sure, as to the first q, nope it should work when the view is clicked on

Reply:
it works without update! changed